apps/bankingAPI/lib/bankingAPI/users/users.ex

```diff
@@ -0,0 +1,53 @@
+defmodule BankingAPI.Users do
+  @moduledoc """
+  Domain public functions for users context
+  """
+
+  alias BankingAPI.Users.Inputs
+  alias BankingAPI.Users.Schemas.User
+  alias BankingAPI.Repo
+
+  require Logger
+
+  @doc """
+  Given a VALID changeset it attempts to insert a new user.
+  It might fail due to email unique index and we transform that return
+  into an error tuple.
+  """
+  @spec create_new_user(Inputs.Create.t()) ::
+          {:ok, User.t()} | {:error, Ecto.Changeset.t() | :email_conflict}
+  def create_new_user(%Inputs.Create{} = input) do
+    Logger.info("Inserting new user")
+
+    params = %{name: input.name, email: input.email}
+
+    with %{valid?: true} = changeset <- User.changeset(params),
+         {:ok, user} <- Repo.insert(changeset) do
+      Logger.info("User successfully inserted. Email: #{user.email}")
+      {:ok, user}
+    else
+      %{valid?: false} = changeset ->
+        Logger.error("Error while inserting new user. Error: #{inspect(changeset)}")
+        {:error, changeset}
+
+      {:error, changeset} ->
+        Logger.error("Error while inserting new user. Error: #{inspect(changeset)}")
```

Comment:
No geral a gente loga coisas variáveis e paramêtros nos metadados do log ao invés de no texto:
```suggestion
        Logger.error("Error while inserting new user", error: inspect(changeset))
```